### DISCUSSION TITLE
.Net 6/7 PublishSingleFile

### DISCUSSION THREAD

**Discussion by phuocminh6464** [cefsharp/CefSharp#4444] [General]
Hi everyone, my WPF app using net 7 and I want to publish app as single EXE file but after publishing I can not open the app. This is command I'm using `dotnet publish -c Release -f net7.0-windows -r win-x64 -p:PublishSingleFile=true` and the error details in Events Viewer.

![aaaa](https://user-images.githubusercontent.com/26184798/229045809-94744777-3148-4287-926e-961a89fcde0c.PNG)

When I open publish folder, the files such as: CefSharp.Core.dll, CefSharp.Core.Runtime.dll, CefSharp.dll, CefSharp.Wpf.dll, etc are not pulished. However, if it remove `-p:PublishSingleFile=true` from the command, opening app is normal and those files exist in publish folder.

What need I do to fix this issue?

**Comment by amaitland** [comment]
Publishing as a single file exe is discussed in https://github.com/cefsharp/CefSharp/issues/3407 

.Net 5/6/7 should all behave the same in terms of settings. The publish args I'd recommend are at https://github.com/cefsharp/CefSharp/issues/3407#issuecomment-787008626

You will need to `SelfHost` the `BrowserSubprocess`, there's more up to date instructions at https://github.com/cefsharp/CefSharp/wiki/SelfHost-BrowserSubProcess

**Reply by phuocminh6464** [reply]
Thanks you so much. I don't know why the command is not working, but if changing to use Publish folder by Right-click on the project  in Solution Explorer => select Publish then it can publish as a single file.
